Release browser hail tables (#1583)
broadinstitute/gnomad-browser: browser/help/topics/v4-browser-hts.md

---
id: v4-browser-hts
title: 'gnomAD v4 Browser Hail Tables'
---

> To go from the gnomAD release tables to a format that is more convenient for the browser to display, the input tables undergo a few transformations. This groups data in ways that are more convenient to query from the database these tables ultimately get loaded into. In addition to performing some reshaping of the data, additional annotations or information is added as needed.

Reviewer: I wonder if we can we add a bit more high level explanation about these tables? How about:

---

We release two additional data tables to the public so that our users can more easily incorporate gnomAD data into external pipelines and analyses in a manner consistent with what they see in the browser.

## gnomAD v4.1 exome/genome variant table

To convert the standard gnomAD variant release tables into a format more suitable for browser display, we join the exome and genome tables on locus/allele to create a single table. This process ensures that they share the same site-level annotations, thus saving space and optimizing database/API queries. Additionally, frequencies are structured in a JSON-like format that is more easily consumable by web applications.

Each row (i.e., variant) in this table will have distinct allele frequency information and quality metrics depending whether it was present in the exome or genome callsets, but will share common annotations such as [VEP annotations](https://useast.ensembl.org/info/docs/tools/vep/index.html) and _in silico_ predictors.

The script for how this table is created can be found [here](
https://github.com/broadinstitute/gnomad-browser/blob/main/data-pipeline/src/data_pipeline/pipelines/gnomad_v4_variants.py).

## gnomAD 4.1 genes table

The second table underlies the gene models data seen in the browser, which contains detailed information on exon-coding regions, transcripts, identifiers, gene constraint, and co-occurrence data. The data from this table is derived from [GENCODE 39](https://www.gencodegenes.org/human/release_39.html), the [HUGO Gene Nomenclature Committee (HGNC)](https://www.genenames.org/), [MANE transcripts](https://www.ncbi.nlm.nih.gov/refseq/MANE/), [GTEx](https://gtexportal.org/home/) (coming soon), and from gnomAD secondary analyses.

The script for how this table is created can be found [here](https://github.com/broadinstitute/gnomad-browser/blob/main/data-pipeline/src/data_pipeline/pipelines/genes.py).

# Browser Hail Table Field Descriptions

#### gnomAD v4.1 browser variant Hail Table annotations

Global fields:

- `mane_transcripts_version`: MANE Select version used to annotate variants.

Row fields:

- `locus`: Variant locus. Contains contig and position information.
- `alleles`: Variant alleles.
- `exome`: Struct containing information about variant from exome data.
  - `colocated_variants`: Struct containing arrays of colocated variants.
  - `all`: An array containing colocated variants that are present in the entire exome dataset.
  - `non_ukb`: An array containing colocated variants that are present in the non-UK Biobank (UKB) subset of the dataset.
  - `subsets`: A set containing the subsets this variant is seen in.

>   - `flags`: A set containing the flags about the region this variant falls in.

Reviewer 2: ```suggestion
  - `flags`: A set containing the flags about the region this variant falls in. See `region_flags` description on v4 Hail Tables [help page](https://gnomad.broadinstitute.org/help/v4-hts).
```

  - `freq`: A struct containing variant frequency information for each subset.
  - `all`: Struct containing variant frequency information calculated across all samples.

>     - `ac`: The alternate allele count for this variant calculated across high-quality genotypes (genotypes with depth >= 10, genotype quality >= 20 and minor allele balance > 0.2 for heterozygous genotypes).

Reviewer 2: ```suggestion
    - `ac`: The alternate allele count for this variant calculated across high-quality genotypes (genotypes with depth >= 10, genotype quality >= 20 and minor allele balance > 0.2 for heterozygous genotypes). This is the allele count displayed in the gnomAD browser (not `ac_raw` below).
```

    - `ac_raw`: The alternate allele count for this variant calculated across unadjusted genotypes.
    - `an`: Total number of alleles for this locus.
    - `hemizygote_count`: Number of hemizygous alternate individuals.
    - `homozygote_count`: Number of homozygous alternate individuals.
    - `ancestry_groups`: Array containing variant frequency information stratified per genetic ancestry group.
      - `id`: Three letter identifier for this genetic ancestry group, e.g. `amr` or `sas`.
      - `ac`: Alternate allele count for this variant in this genetic ancestry group.
      - `an`: Total number of alleles for this locus for this genetic ancestry group.
      - `hemizygote_count`: Number of hemizygous alternate individuals in this genetic ancestry group.
      - `homozygote_count`: Number of homozygous alternate individuals in this genetic ancestry group.
    - `non_ukb`: Struct containing variant frequency information from the non-UKB subset. Includes same fields as above struct (`all`).
  - `faf95`: Struct containing information about the filtered allele frequency (FAF; 95% confidence interval [CI]).

>     - `grpmax`: Genetic ancestry group maximum (groupmax) FAF.

Reviewer 2: ```suggestion
    - `grpmax`: Genetic ancestry group maximum (grpmax) FAF.
```

>     - `grpmax_gen_anc`: Genetic ancestry group associated with groupmax FAF.

Reviewer 2: ```suggestion
    - `grpmax_gen_anc`: Genetic ancestry group associated with grpmax FAF.
```

  - `faf99`: Struct containing information about the FAF (99% CI). Contains same subfields as above (`faf95`).
  - `fafmax`: Struct containing information about the maximum FAF.
    - `gnomad`: Struct containing information about the fafmax for all of gnomad for the joint exome and genome data.
      - `faf95_max`: Max FAF value for the (95% CI).
      - `faf95_max_gen_anc`: Genetic ancestry group associated with the grpmax FAF (95% CI).

>       - `faf99_max`: Max FAF(99% CI).f

Reviewer 2: ```suggestion
      - `faf99_max`: Max FAF (99% CI).
```

      - `alt_adj`: Struct containing DP information calculated using high-quality genotypes (variant carriers only). Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).
      - `alt_raw`: Struct containing DP information calculated across unadjusted genotypes (variant carriers only). Structured to allow ease of creating histograms. Contains same fields as other histogram structs (e.g. `het`)
    - `genotype_quality`: Struct containing information used to display genotype quality (GQ) histograms.
      - `all_adj`: Struct containing GQ information calculated using high-quality genotypes. Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).
      - `all_raw`: Struct containing GQ information calculated across unadjusted genotypes. Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).
      - `alt_adj`: Struct containing GQ information calculated using high-quality genotypes (variant carriers only). Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).

>       - `alt_raw`: Struct containing GQ information calculated across unadjusted genotypes (variant carriers only). Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).

Reviewer 2: these descriptions are informative but a bit wordy (apologies if I suggested this previously). how about replacing
> Structured to allow easy histogram creation. Contains same fields as other histogram structs (e.g., `het`).

in each with just
>  Contains same fields as other histogram structs.

  - `site quality metrics`: Array containing site quality metric information.
    - `metric`: Metric name.
    - `value`: Metric value.
- `genome`: Struct containing information about this variant from genome data. Contains all the same fields as the exome data, with the exception that the subsets are (`all` `hgdp`, `tgp`) instead of (`all`, `non_ukb`).
- `rsids`: dbSNP reference SNP identification (rsID) numbers.

> - `in_silico_predictors`: Variant prediction annotations. Struct contains prediction scores from multiple in silico predictors for variants that are predicted to be missense, impacting protein function, evolutionarily conserved, or splice-altering. We chose scores for either MANE Select or canonical transcripts if a prediction score was available for multiple transcripts.

Reviewer 2: perhaps you could point users to this section (`in_silico_predictors`) on the v4 HTs help page here? would it be helpful to create hyperlinks to sections of the v4 HTs help page that contain fields that are also described here (e.g., `region_flags`, `in_silico_predictors`, `filters`)?

  - `phylop`: Base-wise conservation [score](https://pubmed.ncbi.nlm.nih.gov/37104612/) across the 241 placental mammals in the Zoonomia project. Score ranges from -20 to 9.28, and reflects acceleration (faster evolution than expected under neutral drift, assigned negative scores) as well as conservation (slower than expected evolution, assigned positive scores).
  - `sift_max`: [Score](https://www.nature.com/articles/nprot.2009.86) reflecting the scaled probability of the amino acid substitution being tolerated, ranging from 0 to 1. Scores below 0.05 are predicted to impact protein function. We prioritize max scores for MANE Select transcripts where possible and otherwise report a score for the canonical transcript.
  - `polyphen_max`: [Score](https://www.nature.com/articles/nmeth0410-248) that predicts the possible impact of an amino acid substitution on the structure and function of a human protein, ranging from 0.0 (tolerated) to 1.0 (deleterious). We prioritize max scores for MANE Select transcripts where possible and otherwise report a score for the canonical transcript.

> - `variant_id`: gnomAD variant ID.- `faf95_joint`: A struct containing joint (exome + genome) FAF information (95% CI).

Author: Ah, this field `faf95_joint` is the one thats its own seperate struct outside of `exome` or `genome`. The `genome` struct just has some placeholder information saying "pretty much same exact shape as the exomes above".

I've made some indentation mistakes that have definitely made this more confusing, thanks for catching those.

  - `grpmax`: Groupmax FAF value for all genetic ancestry groups across exomes + genomes.
  - `grpmax_gen_anc`: Genetic ancestry group associated with the value `grpmax` above.
- `faf99_joint`: A struct containing joint (exome + genome) FAF (99% CI). - `colocated_variants`: Array containing variants that are located at the same locus as this variant.
- `coverage`: Struct containing coverage information for locus.

>   - `exome`: Struct containing exome coverage information.

Reviewer 2: should the fields beneath this (from `mean` to `over_100`) be indented?

  - `mean`: Mean depth of coverage at this locus.
  - `median`: Median depth of coverage at this locus.
  - `over_1`: Percentage of samples with a coverage greater than 1 at this locus.

>   - `over_5`: TPercentage of samples with a coverage greater than 5 at this locus

Reviewer 3: Just flagging typo here.

  - `over_10`: Percentage of samples with a coverage greater than 10 at this locus
  - `over_15`: Percentage of samples with a coverage greater than 15 at this locus.
  - `over_20`: Percentage of samples with a coverage greater than 20 at this locus
  - `over_25`: Percentage of samples with a coverage greater than 25 at this locus.
  - `over_30`: Percentage of samples with a coverage greater than 30 at this locus.
  - `over_50`: Percentage of samples with a coverage greater than 50 at this locus.
  - `over_100`: Percentage of samples with a coverage greater than 100 at this locus.
  - `genome`: Struct containing genome coverage information. Contains the same fields as `exome` above.

>   - `transcript_consequences`: Array containing variant transcript consequence information.

Reviewer 2: should the fields beneath this be indented as well?

  - `biotype`: Transcript biotype.
  - `consequence_terms`: Array of predicted functional consequences.
  - `domains`: Sset containing protein domains affected by variant.
  - `gene_id`: Unique ID of gene associated with transcript.
  - `hgvsc`: HGVS coding sequence notation for variant.
  - `hgvsp`: HGVS protein notation for variant.
  - `is_canonical`: Whether transcript is the canonical transcript.
  - `lof_filter`: Variant LoF filters (from [LOFTEE](https://github.com/konradjk/loftee)).
  - `lof_flags`: LOFTEE flags.
  - `lof`: Variant LOFTEE status (high confidence [HC] or low confidence [LC]).
  - `major_consequence`: Primary consequence associated with transcript.
  - `transcript_id`: Unique transcript ID.
  - `transcript_version`: Transcript version.
  - `gene_version`: Gene version.
  - `is_mane_select`: Whether transcript is the MANE select transcript.

>   - `is_mane_select_version`: MANE Select version, has a value if this transcript is the MANE select transcript.

Reviewer 2: ```suggestion
  - `is_mane_select_version`: MANE Select version; field has a value if this transcript is the MANE select transcript.
```

  - `refseq_id`: RefSeq ID associated with transcript.
  - `refseq_version`: RefSeq version.

#### gnomAD v4.1. browser gene models Hail Table annotations

> Row fields:

Reviewer 2: just checking that the globals are removed after full pipeline has been run -- I see this in `gs://gnomad-v4-data-pipeline/output/genes/genes_grch38_annotated_5.ht`:
```
----------------------------------------
Global fields:
    'annotations': struct {
        'canonical_transcript': struct {

        }
        'mane_select_transcript': struct {
            'version': str
        }
    }
----------------------------------------
```

Reviewer 2: sorry, I worded that comment poorly -- it might be helpful to keep this global
```
- `mane_select_version`: MANE Select version used to annotate variants.
```
on this table as well (since this specifically contains gene information)

- `interval`: Struct representing start and end positions of gene.

> - `gene_id`: Unique gene ID.

Reviewer 2: this is the ensembl gene ID, right? (might be helpful to include)

- `gene_version`: Gene version.
- `gencode_symbol`: GENCODE gene symbol.
- `chrom`: Chromosome in which gene is located.

> - `strand`:Gene strand.

Reviewer 3: I think another typo.

- `start`: Gene genomic start position (position only).

> - `stop`: Gene genomic stop position (position only.- `xstart`: Gene genomic start position (format: chromosomeposition).Note that chrX is encoded as 23, chrY as 24, and chrM as 25.

Reviewer 2: ```suggestion
- `stop`: Gene genomic stop position (position only).
- `xstart`: Gene genomic start position (format: chromosomeposition). Note that chrX is encoded as 23, chrY as 24, and chrM as 25.
```
this is pretty self explanatory, but it might be nice to have an example here as well

  - `interval`: Struct representing the start and end positions of transcript.
  - `transcript_id`: Unique transcript ID.
  - `transcript_version`: Transcript version.
  - `gene_id`: Unique gene ID.
  - `gene_version`: Gene version.
  - `chrom`: Chromosome in which transcript is located.
  - `strand`: Transcript strand.
  - `start`: Transcript genomic start position (position only).
  - `stop`: Transcript genomic stop position (position only).
  - `xstart`: Transcript genomic start position (format: chromsomeposition).
  - `xstop`: Transcript genomic stop position (format: chromsomeposition).
  - `exons`: Array containing transcript exon information.
  - `feature_type`: Exon type (e.g., CDS).
  - `start`: Exon genomic start position (position only).
  - `stop`: Exon genomic stop position (position only).
  - `xstart`: Exon genomic start position (format: chromsomeposition).
  - `xstop`: Exon genomic start position (format: chromsomeposition).
  - `reference_genome`: Reference genome associated with this transcript.
  - `refseq_id`: Transcript RefSeq identifier.
  - `refseq_version`: RefSeq version.

> - `hgnc_id`: GeneHGNC identifier.

Reviewer 2: ```suggestion
- `hgnc_id`: HGNC gene identifier.
```

- `symbol`: Gene symbol.
- `name`: Gene name.
- `previous_symbols`: Set containing previous gene symbols.
- `alias_symbols`: Set containing alternate gene symbols.
- `omim_id`: Gene OMIM ID.
- `ncbi_id`: Gene NCBI ID.
- `symbol_upper_case`: All-caps gene symbol.
- `search_terms`: Set containing search terms associated with gene.
- `reference_genome`: Reference genome build associated with this gene.
- `flags`: Set containing gene flags this gene.
- `canonical_transcript_id`: Canonical transcript ID.
- `mane_select_transcript`: Struct containing MANE Select transcript information.
  - `matched_gene_version`: Version of the matched gene.
  - `ensembl_id`: Transcript Ensembl ID.
  - `ensembl_version`: Ensembl version.
  - `refseq_id`: Transcript RefSeq ID.
  - `refseq_version`: RefSeq version.

> - `preferred_transcript_id`: Transcript shown on the gene page by default. Field containsMANE Select transcript ID if it exists, otherwise contains Ensemblcanonical transcript ID.

Reviewer 2: ```suggestion
- `preferred_transcript_id`: Transcript shown on the gene page by default. Field contains MANE Select transcript ID if it exists, otherwise contains Ensembl canonical transcript ID.
```

> - `preferred_transcript_source`: Source of transcript ID used for `preferred_transcript_id` field; either "`mane_select`" or "`ensemble_canonical`".

Reviewer 2: ```suggestion
- `preferred_transcript_source`: Source of transcript ID used for `preferred_transcript_id` field; either "`mane_select`" or "`ensembl_canonical`".
```

  - `exp_syn`: Expected number of rare synonymous variants.
  - `obs_lof`: Observed number of rare loss-of-function variants.
  - `obs_mis`: Observed number of rare missense variants.
  - `obs_syn`: Observed number of rare synonymous variants.
  - `oe_lof`: Observed/expected (OE) ratio for rare loss-of-function variants.
  - `oe_lof_lower`: Lower bound of ratio for rare loss-of-function variants.
  - `oe_lof_upper`: Upper bound of the OE ratio (LOEUF) for rare loss-of-function variants.
  - `oe_mis`: OE ratio for rare missense variants.
  - `oe_mis_lower`: Lower bound of OE ratio for rare missense variants.
  - `oe_mis_upper`: Upper bound of OE ratio for rare missense variants.
  - `oe_syn`: OE ratio for rare synonymous variants.
  - `oe_syn_lower`: Lower bound of OE ratio for rare synonymous variants.
  - `oe_syn_upper`: Upper bound of the OE ratio for rare synonymous variants.
  - `lof_z`: Z-score for rare loss-of-function variants.
  - `mis_z`: Z-score for rare missense variants.
  - `syn_z`: Z-score for rare synonymous variants.
  - `pli`: Probability of being loss-of-function intolerant (pLI) score.
  - `flags`: Set containing constraint flags for transcript.

> - `heterozygous_variant_cooccurrence_counts`: Array containing information about heterozygous variant co-occurrence counts.

Reviewer 2: ```suggestion
- `heterozygous_variant_cooccurrence_counts`: Array containing information about heterozygous variant co-occurrence counts. Struct is only present on the GRCh37 Hail Table.
```

  - `csq`: Variant consequence.
  - `af_cutoff`: Allele frequency cutoff.
  - `data`: Struct containing variant co-occurrence data.
  - `in_cis`: Count of variants in cis.
  - `in_trans`: Count of variants in trans.
  - `unphased`: Count of unphased variants.
  - `two_het_total`: Total count of two heterozygous variants.

> - `homozygous_variant_cooccurrence_counts`: Array containing information about homozygous variant co-occurrence counts.

Reviewer 2: ```suggestion
- `homozygous_variant_cooccurrence_counts`: Array containing information about homozygous variant co-occurrence counts. Struct is only present on the GRCh37 Hail Table.
```

      - `an`: Total number of alleles for this locus for this genetic ancestry group.
      - `hemizygote_count`: Number of hemizygous alternate individuals in this genetic ancestry group.
      - `homozygote_count`: Number of homozygous alternate individuals in this genetic ancestry group.
    - `non_ukb`: Struct containing variant frequency information from the non-UKB subset. Includes same fields as above struct (`all`).
  - `faf95`: Struct containing information about the filtered allele frequency (FAF; 95% confidence interval [CI]).
    - `grpmax`: Genetic ancestry group maximum (grpmax) FAF.
    - `grpmax_gen_anc`: Genetic ancestry group associated with grpmax FAF.
  - `faf99`: Struct containing information about the FAF (99% CI). Contains same subfields as above (`faf95`).
  - `fafmax`: Struct containing information about the maximum FAF.
    - `gnomad`: Struct containing information about the fafmax for all of gnomad for the exome data.
      - `faf95_max`: Max FAF value for the (95% CI).
      - `faf95_max_gen_anc`: Genetic ancestry group associated with the grpmax FAF (95% CI).
      - `faf99_max`: Max FAF(99% CI).

>       - `faf99_max_gen_anc`: Genetic ancestry group associated with the max FAF (99% CI).

Author: Yeah, this was a bit of a last minute transformation we did to allow our API code to look for the same fields on the v2 dataset as on the v4 dataset, so its currently duplicated.

I can make an issue that logs all the odd duplications like this to remove before the next big dataload, if that works? Alternatively we could temporarily put in a last step that cleans up this table for release, and have an associated issue to clean it up in th earlier steps.

>       - `faf99_max_gen_anc`: Genetic ancestry group associated with the max FAF (99% CI).

Reviewer 2: is it messy to do the second option (temporarily adding a step to clean up the table for release + creating a second issue to update the pipeline to clean up duplicate fields upstream)? removing duplicate fields will make this documentation clearer for the users, and we ideally won't store duplicate fields in a future release (e.g., in v5).

that said, I want to make sure this second option isn't too unbalanced in terms of effort -- is this second option also significantly more work?

>       - `faf99_max_gen_anc`: Genetic ancestry group associated with the max FAF (99% CI).

Author: Sweet, thanks for the thoughts, we'll go with the second option. It'll make for a better table in the short and long term.

      - `alt_raw`: Struct containing GQ information calculated across unadjusted genotypes (variant carriers only). Contains same fields as other histogram structs.
  - `site quality metrics`: Array containing site quality metric information.
    - `metric`: Metric name.
    - `value`: Metric value.
- `genome`: Struct containing information about this variant from genome data. Contains all the same fields as the exome data, with the exception that the subsets are (`all` `hgdp`, `tgp`) instead of (`all`, `non_ukb`).
- `rsids`: dbSNP reference SNP identification (rsID) numbers.
- `in_silico_predictors`: Variant prediction annotations. Struct contains prediction scores from multiple in silico predictors. See `in_silico_predictors` description on the v4 Hail Tables [help page](v4-hts#in-silico-predictors)
- `variant_id`: gnomAD variant ID.
- `faf95_joint`: A struct containing joint (exome + genome) FAF information (95% CI).
  - `grpmax`: Groupmax FAF value for all genetic ancestry groups across exomes + genomes.
  - `grpmax_gen_anc`: Genetic ancestry group associated with the value `grpmax` above.
- `faf99_joint`: A struct containing joint (exome + genome) FAF (99% CI).

> - `colocated_variants`: Array containing variants that are located at the same locus as this variant.

Author: Yep, will add a few words clarifying that.

- `omim_id`: Gene OMIM ID.
- `ncbi_id`: Gene NCBI ID.
- `symbol_upper_case`: All-caps gene symbol.
- `search_terms`: Set containing search terms associated with gene.
- `reference_genome`: Reference genome build associated with this gene.
- `flags`: Set containing gene flags this gene.
- `canonical_transcript_id`: Canonical transcript ID.
- `mane_select_transcript`: Struct containing MANE Select transcript information.
  - `matched_gene_version`: Version of the matched gene.
  - `ensembl_id`: Transcript Ensembl ID.
  - `ensembl_version`: Ensembl version.
  - `refseq_id`: Transcript RefSeq ID.
  - `refseq_version`: RefSeq version.
- `preferred_transcript_id`: Transcript shown on the gene page by default. Field contains MANE Select transcript ID if it exists, otherwise contains Ensembl canonical transcript ID.
- `preferred_transcript_source`: Source of transcript ID used for `preferred_transcript_id` field; either "`mane_select`" or "`ensembl_canonical`".
- `gnomad_constraint`: Struct containing gnomAD constraint information for gene. Struct is only present on the GRCh37 Hail Table.
  - `gene`: Gene name.
  - `transcript`: Transcript ID.

>   - `mane_select`: Whether transcript is the MANE Select transcript.

Author: The v2 version (GRCh37 gene models) currently has gnomAD v2 constraint annotated.

A step was added to remove constraint from the v4 (GRCh38 gene models) due to the constraint being a bit more experimental.

Should we also remove constraint from the GRCh37/v2 gene model Hail Table?

>   - `mane_select`: Whether transcript is the MANE Select transcript.

Reviewer 2: oh sorry I wasn't clear -- I meant that v2 constraint doesn't have `mane_select` annotated iirc (happy to be corrected, though). From my perspective, v2 gene constraint is stable and won't change, and it seems valuable to include in the gene models tables for our users (so fine with me to keep gene constraint in the GRCh37 table but remove from GRCh38 table)

>   - `mane_select`: Whether transcript is the MANE Select transcript.

Author: Ohhh I see oops I misunderstood. 

Good catch, will correct the fields here.

      - `all_adj`: Struct containing DP information calculated using high-quality genotypes. Contains same fields as other histogram structs.
      - `all_raw`: Struct containing DP information calculated across unadjusted genotypes. Contains same fields as other histogram structs.
      - `alt_adj`: Struct containing DP information calculated using high-quality genotypes (variant carriers only). Contains same fields as other histogram structs.
      - `alt_raw`: Struct containing DP information calculated across unadjusted genotypes (variant carriers only). Contains same fields as other histogram structs.
    - `genotype_quality`: Struct containing information used to display genotype quality (GQ) histograms.
      - `all_adj`: Struct containing GQ information calculated using high-quality genotypes. Contains same fields as other histogram structs.
      - `all_raw`: Struct containing GQ information calculated across unadjusted genotypes. Contains same fields as other histogram structs.
      - `alt_adj`: Struct containing GQ information calculated using high-quality genotypes (variant carriers only). Contains same fields as other histogram structs.

>       - `alt_raw`: Struct containing GQ information calculated across unadjusted genotypes (variant carriers only). Contains same fields as other histogram structs.

Reviewer 2: on line 94 below, you have this description:
```
See `histograms` description on the v4 Hail Tables [help page](v4-hts#joint-histograms).
```
you could apply that here as well to save space

- `xstop`: Gene genomic stop position (format: chromosomeposition).
- `exons`: Array containing exon information for gene.
  - `feature_type`: Exon type (e.g., CDS).
  - `start`: Exon genomic start position (position only).
  - `stop`: Exon genomic stop position (position only).
  - `xstart`: Exon genomic start position (format: chromosomeposition).
  - `xstop`: Exon genomic stop position (format: chromosomeposition).
- `transcripts`: Array containing information about transcripts associated with the gene.
  - `interval`: Struct representing the start and end positions of transcript.
  - `transcript_id`: Unique transcript ID.
  - `transcript_version`: Transcript version.
  - `gene_id`: Unique gene ID.
  - `gene_version`: Gene version.
  - `chrom`: Chromosome in which transcript is located.
  - `strand`: Transcript strand.
  - `start`: Transcript genomic start position (position only).
  - `stop`: Transcript genomic stop position (position only).

>   - `xstart`: Transcript genomic start position (format: chromsomeposition).
>   - `xstop`: Transcript genomic stop position (format: chromsomeposition).

Reviewer 2: ```suggestion
  - `xstart`: Transcript genomic start position (format: chromosomeposition).
  - `xstop`: Transcript genomic stop position (format: chromosomeposition).
```
sorry, I must have missed these previously!

  - `exons`: Array containing transcript exon information.
  - `feature_type`: Exon type (e.g., CDS).
  - `start`: Exon genomic start position (position only).
  - `stop`: Exon genomic stop position (position only).

>   - `xstart`: Exon genomic start position (format: chromsomeposition).
>   - `xstop`: Exon genomic start position (format: chromsomeposition).

Reviewer 2: ```suggestion
  - `xstart`: Exon genomic start position (format: chromosomeposition).
  - `xstop`: Exon genomic start position (format: chromosomeposition).
```

  - `reference_genome`: Reference genome associated with this transcript.
  - `refseq_id`: Transcript RefSeq ID.
  - `refseq_version`: RefSeq version.
- `hgnc_id`: HGNC gene ID.
- `symbol`: Gene symbol.
- `name`: Gene name.
- `previous_symbols`: Set containing previous gene symbols.
- `alias_symbols`: Set containing alternate gene symbols.
- `omim_id`: Gene OMIM ID.
- `ncbi_id`: Gene NCBI ID.
- `symbol_upper_case`: All-caps gene symbol.
- `search_terms`: Set containing search terms associated with gene.
- `reference_genome`: Reference genome build associated with this gene.

> - `flags`: Set containing gene flags this gene.

Reviewer 2: ```suggestion
- `flags`: Set containing gene flags for this gene.
```
(or something else; please correct if my suggestion is wrong)
